### ISSUE TITLE
Vanilla Dropdown's use of `keydown`

### ISSUE THREAD

**Issue by elizabethsjudd** [carbon-design-system/carbon#4874]
## Environment

> Operating system

Mac

> Browser

Safari, Chrome, FF

> Automated testing tool and ruleset

2.5.2 Pointer Cancellation. For functionality that can be operated using a single pointer, completion of the function is on the up-event with an ability to abort, undo or reverse the outcome. (Level A)

This is a new WCAG 2.1 feature

> Assistive technology used to verify

Keyboard

## Detailed description

> What version of the Carbon Design System are you using?

10.7.3

> What did you expect to happen?

keyboard commands happen on `keyup` not `keydown`

> What happened instead?

keyboard commends are implemented for `keydown`

> What WCAG 2.1 checkpoint does the issue violate?

2.5.2

## Steps to reproduce the issue

https://github.com/carbon-design-system/carbon/blob/master/packages/components/src/components/dropdown/dropdown.js#L49

## Additional information

- Can assign the issue to me and we can make the update since Vanilla is not a priority for Carbon right now.  This is a blocker for our release.

**Comment by asudoh** [comment]
Thanks @elizabethsjudd for your interest in jumping in. `mousedown` vs. `mouseup`, etc. tends to be sensitive wrt the coordination with other events, and thus I'd like to mention that the development/review may take time.
